docs/reference/indices/shrink-index.asciidoc

```diff
@@ -227,3 +227,15 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=timeoutparms]
 include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=target-index-aliases]
 
 include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=target-index-settings]
+
+`max_single_primary_size`::
+(Optional, <<byte-units, byte units>>)
+The max single primary shard size of the target index, it's used to find an optimum shards number of the target index.
+When this parameter is set properly, each shard's storage of the target index will not be greater than the parameter,
+while the shards number of the target index still be a factor of the source index's shards number, but if the parameter
+is less than the single shard size of the source index, the shards number of the target index will be equal to the source index's shards number.
+For example, when this parameter is set to `50GB`, if the source index has `60` primary shards with `100GB` storage, then the
+target index will have `2` primary shards, each shard's storage is `50GB`; if the source index has `60` primary shards
+with `1000GB` storage, then the target index will have `20` primary shards; if the source index has `60` primary shards
+with `4000GB` storage, then the target index will still have `60` primary shards. This parameter is conflict
+with `number_of_shards` in the `settings` parameter, either of them can be set.
```

Comment:
For this I think you can copy the changes made in the previous doc (rather than duplicate them here)